src/Groups/matrices/linear_isconjugate.jl

```diff
@@ -0,0 +1,155 @@
+# TODO: in this file are used many methods TEMPORARILY defined in files matrix_manipulation.jl and stuff_field_gen.jl
+# once methods in those files will be deleted / replaced / modified, this file need to be modified too
+
+
+
+export
+    generalized_jordan_block,
+    generalized_jordan_form,
+    isconjugate,
+    issemisimple,
+    isunipotent,
+    pol_elementary_divisors,
+    multiplicative_jordan_decomposition
+
+
+
+
+########################################################################
+#
+# Semisimple / Unipotent
+#
+########################################################################
+
+"""
+    multiplicative_jordan_decomposition(M::MatrixGroupElem)
+Return `S` and `U` in the group `G = parent(M)` such that `S` is semisimple, `U` is unipotent and  `M = SU = US`.
+!!! warning "WARNING:" 
+    this is *NOT*, in general, the same output returned when `M` has type `MatElem`.
+"""
+function multiplicative_jordan_decomposition(x::MatrixGroupElem)
+   a = order(x)
+   p = characteristic(base_ring(x))
+   alpha = valuation(a,p)
+   m = div(a, p^alpha)
+   k = crt(fmpz(0),fmpz(p^alpha),fmpz(1),fmpz(m))
+#   a,b = multiplicative_jordan_decomposition(x.elm)
+   return x^k, x^(a+1-k)
+end
+
+"""
+    issemisimple(x::MatrixGroupElem)
+Return whether `x` is semisimple, i.e. has order coprime with the characteristic of its base ring.
+"""
+issemisimple(x::MatrixGroupElem) = iscoprime(Int(order(x)::fmpz), Int(characteristic(x.parent.ring)))
+
+"""
+    isunipotent(x::MatrixGroupElem)
+Return whether `x` is unipotent, i.e. its order is a power of the characteristic of its base ring.
```

Comment:
Those are not the general descriptions of being semisimple or unipotent; these hold at most over finite fields (obviously). 

So perhaps the method signature can be restricted to `MatrixGroupElem{T} where T<:FinFieldElem` or something like that?

Reply:
Done it.
Though, I think the concept of semisimple and unipotent matrix works for matrices over infinite fields too. The definition is not given in the same terms, of course, since the order may be infinite.

Reply:
The concepts of unipotent and semisimple elements of course exist over infinite fields (I'd argue they come from there), but the definition you give does not work there; rather, I'd usually give your definition as the result of a lemma that characterize such elements over finite fields. Hence the request to restrict the method signature -- we can provide more generic methods for other fields later, of course